code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -941,6 +941,27 @@ func (sc *StorageSmartContract) populateGenerateChallenge(
 		return nil, actErr
 	}
 
+	actErr = cstate.WithActivation(balances, "athena", func() error { return nil }, func() error {
+		blobber, err := sc.getBlobber(blobberID, balances)
+		if err != nil {
+			return common.NewError("add_challenge", err.Error())
+		}
+
+		if blobber.IsKilled() || blobber.IsShutDown() {
+			err := partitionsChallengeReadyBlobbersRemove(balances, blobberID)
```

Comment:
this won't work, since we return error, and when we return error we discard all MPT changes